client/analytics/report/coupons/table.js

```diff
@@ -0,0 +1,227 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, _n } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { format as formatDate } from '@wordpress/date';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { get, map, orderBy } from 'lodash';
+
+/**
+ * WooCommerce dependencies
+ */
+import {
+	appendTimestamp,
+	getCurrentDates,
+	getIntervalForQuery,
+	getDateFormatsForInterval,
+} from '@woocommerce/date';
+import { Link, TableCard } from '@woocommerce/components';
+import { formatCurrency, getCurrencyFormatDecimal } from '@woocommerce/currency';
+import { onQueryChange } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import ReportError from 'analytics/components/report-error';
+import { QUERY_DEFAULTS } from 'store/constants';
+import { getReportChartData, getFilterQuery } from 'store/reports/utils';
+
+class CouponsReportTable extends Component {
+	constructor( props ) {
+		super( props );
```

Comment:
Since we aren't binding or setting any default state, do we need the constructor?

Reply:
Good catch! I removed it also from the Orders table, which I used as an example to build the Coupons table (eb435e150717918d025bacd9ca7685457dac1d37).